pkg/restic/backup.go

```diff
@@ -72,15 +59,197 @@ func (w *ResticWrapper) RunBackup(backupOption BackupOptions) (*BackupOutput, er
 	}
 
 	// Extract information from output of "stats" command
-	err = backupOutput.extractStatsInfo(out)
+	repoSize, err := extractStatsInfo(out)
 	if err != nil {
 		return nil, err
 	}
+	backupOutput.RepositoryStats.Size = repoSize
 
 	// Backup complete. Read current time and calculate total backup duration.
 	endTime := time.Now()
-	backupOutput.HostBackupStats.Duration = endTime.Sub(startTime).String()
-	backupOutput.HostBackupStats.Phase = api_v1beta1.HostBackupSucceeded
+	for idx := range backupOutput.HostBackupStats {
+		if backupOutput.HostBackupStats[idx].Hostname == backupOption.Host {
+			backupOutput.HostBackupStats[idx].Duration = endTime.Sub(startTime).String()
+			backupOutput.HostBackupStats[idx].Phase = api_v1beta1.HostBackupSucceeded
+		}
+	}
 
 	return backupOutput, nil
 }
+
+// RunParallelBackup runs multiple backup in parallel.
+// Host must be different for each backup.
+func (w *ResticWrapper) RunParallelBackup(backupOptions []BackupOptions, maxConcurrency int) (*BackupOutput, error) {
+
+	// Initialize restic repository if it does not exist
+	_, err := w.initRepositoryIfAbsent()
+	if err != nil {
+		return nil, err
+	}
+
+	// WaitGroup to wait until all go routine finish
+	wg := sync.WaitGroup{}
+	// concurrencyLimiter channel is used to limit maximum number simultaneous go routine
+	concurrencyLimiter := make(chan bool, maxConcurrency)
+	defer close(concurrencyLimiter)
+
+	var backupErr error
+	mu := sync.Mutex{} // use lock to avoid racing condition
+	backupOutput := &BackupOutput{}
+
+	for _, opt := range backupOptions {
```

Comment:
The classic range pointer mistake. Use this:

`for i := range backupOptions`

Reply:
Not required. Here, we are sending opt as parameter so no pointer issue.